core/src/main/java/org/apache/carbondata/core/metadata/schema/table/column/ColumnSchema.java

```diff
@@ -126,8 +126,16 @@
    */
   private String aggFunction = "";
 
+  /**
+   * list of parent column relations
+   */
   private List<ParentColumnTableRelation> parentColumnTableRelations;
 
+  /**
+   * timeseries function applied on column
+   */
+  private String timeSeriesFunction = "";
```

Comment:
I think we should create one class to hold all information needed for pre-agg table, instead of putting them here directly in ColumnSchema

Reply:
it is specific to column schema and it has only two property timeseriesfunction and aggregation type

Reply:
@kumarvishal09 I don't think it is required to add timeSeriesFunction here, just rename aggFunction to function and use the same for both aggFunction and timeseries function.